rpc/jsoncodec/codec.go

```diff
@@ -74,21 +75,22 @@ type outMsgV1 struct {
 	ErrorCode string                 `json:"error-code,omitempty"`
 	ErrorInfo map[string]interface{} `json:"error-info,omitempty"`
 	Response  interface{}            `json:"response,omitempty"`
+	TraceID   string                 `json:"trace-id,omitempty"`
+	SpanID    string                 `json:"span-id,omitempty"`
 }
 
+// Close closes the underlying connection and sets the codec to
+// closing mode, so that interface{} further errors are ignored.
 func (c *Codec) Close() error {
-	c.mu.Lock()
```

Comment:
Probably just need to keep an eye on the race tests as there might of been something hanging off these fences.